Deprecate certificate

Jimbo4350 opened the pull request:
Depends on: https://github.com/IntersectMBO/cardano-api/pull/962

# Changelog

```yaml
- description: |
    Deprecate certificate
  type:
  - compatible     # the API has changed but is non-breaking
```

# Context

Additional context for the PR goes here. If the PR fixes a particular issue please provide a [link](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword=) to the issue.

# How to trust this PR

Highlight important bits of the PR that will make the review faster. If there are commands the reviewer can run to observe the new behavior, describe them.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Self-reviewed the diff

<!--
### Note on CI ###
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges. Please contact IOG node developers to do this
for you.
-->

Jimbo4350 committed:
Update friendly rendering of certificates

Jimbo4350 committed:
Propagate new api's experimental certificate type

carbolymer commented on cardano-cli/test/cardano-cli-golden/files/golden/governance/stakeaddress/poolAndNoConfidenceDeleg.cert:
```diff
@@ -1,5 +1,5 @@
 {
-    "type": "CertificateConway",
+    "type": "Certificate",
```
We should make sure that `cardano-testnet` and `cardano-node-tests` pass with this change. I suppose the change to `legacyComparison` here https://github.com/IntersectMBO/cardano-api/pull/962/files should do the job.

Jimbo4350 replied:
The change here is due to the usage of the new `Certificate` type. The `HasTextEnvelope` instance no longer contains the era in its definition:
```
instance
  ( Typeable ledgerera
  , L.EraTxCert ledgerera
  )
  => HasTextEnvelope (Certificate ledgerera)
  where
  textEnvelopeType _ = "Certificate"
```

Jimbo4350 replied:
When reading a certificate with `"CertificateConway"` as its `type` then `legacyComparison` will take over and it will be deserialized.

carbolymer reviewed (commented):
LGTM. I'll approve after we deal with CLI version for node 10.6

Jimbo4350 committed:
Certificate deprecation changes

Jimbo4350 committed:
Bump CHaP index and bump to cardano-api-10.20.0.0

Jimbo4350 committed:
Bump nix flake

Jimbo4350 committed:
Use legacyComparison

Jimbo4350 committed:
Replace checkTextEnvelopeFormat with diffFileVsGoldenFile

Jimbo4350 committed:
Update golden files as we are using the new Certificate type

Jimbo4350 committed:
The genesis key delegation certificate test was never a golden test
Convert this test to a simple certificate decoding test and move the
test to cardano-cli-test
